### ISSUE TITLE
libspatialindex install fails: aclocal not found

### ISSUE THREAD

**Issue by tkarna** [firedrakeproject/firedrake#779]
Running `firedrake-update` on Ubuntu 15.10 dies with:

```
Running aclocal
./autogen.sh: 37: ./autogen.sh: aclocal: not found

  Something went wrong, giving up!

Traceback (most recent call last):
  File "../bin/firedrake-update", line 970, in <module>
    build_and_install_libspatialindex()
  File "../bin/firedrake-update", line 607, in build_and_install_libspatialindex
    check_call(["./autogen.sh"])
  File "../bin/firedrake-update", line 205, in check_call
    subprocess.check_call(arguments, env=env)
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['./autogen.sh']' returned non-zero exit status 1
```

Installing
```
sudo apt-get install automake
```
seems to resolve the issue.

**Comment by miklos1** [comment]
We could probably just push configure to our libspatialindex repository and skip the `./autogen.sh` call.

**Comment by miklos1** [comment]
I have tried putting `configure` in the repository and skipping `./autogen.sh`, but `configure` somehow still wants to call `aclocal`. Even worse, in that case `configure` tried to call a specific version of `aclocal` which causes failure on Travis, while the `./autogen.sh` + `./configure` just works on Travis.

Unless someone knows better how to drive autotools, I suggest we just swallow it and document the dependency on `automake`.

**Comment by dham** [comment]
I think that's fine. automake is not very exotic.

**Comment by stephankramer** [comment]
We had the same issue in fluidity. If you add [AM_MAINTAINER_MODE](https://www.gnu.org/software/automake/manual/html_node/maintainer_002dmode.html) at the start of configure.ac. It will stop ./configure+make using this hideous boot-strapping process where it tries to rebuild configure and the Makefile.in's themselves based on timestamps (which often get messed up by git). Also we found on some systems even with a complete automake toolchain it would not rebuild these files correctly.

**Comment by miklos1** [comment]
I think I managed to implement Stephan's suggestion in the branch `libspatialindex-dist`.

**Comment by miklos1** [comment]
Hopefully fixed by #783.
